File: Backend/internal/dto/guide_course.go
package dto

import "encoding/json"

type CreateGuideCourseRequest struct {
	Slug        string          `json:"slug" binding:"required,min=1,max=100"`
	Title       string          `json:"title" binding:"required,min=1,max=255"`
	Description string          `json:"description" binding:"max=5000"`
	Children    json.RawMessage `json:"children" binding:"required"`
	SortOrder   int             `json:"sort_order"`
	IsPublished *bool           `json:"is_published"`
}

type UpdateGuideCourseRequest struct {
	Slug        string          `json:"slug" binding:"required,min=1,max=100"`
	Title       string          `json:"title" binding:"required,min=1,max=255"`
	Description string          `json:"description" binding:"max=5000"`
	Children    json.RawMessage `json:"children" binding:"required"`
	SortOrder   int             `json:"sort_order"`
	IsPublished bool            `json:"is_published"`
}

type GuideCourseResponse struct {
	ID            string          `json:"id"`
	Title         string          `json:"title"`
	Description   string          `json:"description"`
	Children      json.RawMessage `json:"children,omitempty"`
	LessonCount   int             `json:"lessonCount,omitempty"`
	SectionCount  int             `json:"sectionCount,omitempty"`
	SortOrder     int             `json:"sortOrder,omitempty"`
	IsPublished   bool            `json:"isPublished,omitempty"`
	CreatedAt     string          `json:"createdAt,omitempty"`
	UpdatedAt     string          `json:"updatedAt,omitempty"`
}

type GuideCourseListResponse struct {
	Data []GuideCourseResponse `json:"data"`
}

type GuideCourseBreadcrumbItem struct {
	ID    string `json:"id"`
	Title string `json:"title"`
	Href  string `json:"href"`
}

type GuideCourseSummary struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
}

type GuideLessonResponse struct {
	Lesson     json.RawMessage             `json:"lesson"`
	Breadcrumb []GuideCourseBreadcrumbItem `json:"breadcrumb"`
	Course     GuideCourseSummary          `json:"course"`
}

File: Admin/src/types/guideCourse.ts
import type { DeltaContent } from './delta'

export type { DeltaContent, DeltaOp } from './delta'

export type LessonBlock =
  | { id: string; type: 'title'; text: string }
  | { id: string; type: 'description'; delta: DeltaContent }
  | { id: string; type: 'video'; src: string; poster?: string; caption?: string }
  | { id: string; type: 'image'; src: string; alt?: string; caption?: string }
  | { id: string; type: 'link'; href: string; label: string; description?: string }
  | {
      id: string
      type: 'file'
      url: string
      title: string
      ext: string
      sizeLabel: string
      description?: string
    }

export interface LessonNode {
  kind: 'dars'
  id: string
  title: string
  subtitle?: string
  durationMin?: number
  blocks: LessonBlock[]
}

export interface SectionNode {
  kind: 'bolim'
  id: string
  title: string
  children: LessonNode[]
}

export type CourseNode = LessonNode | SectionNode

export interface GuideCourse {
  id: string
  title: string
  description: string
  children?: CourseNode[]
  lessonCount?: number
  sectionCount?: number
  sortOrder?: number
  isPublished?: boolean
  createdAt?: string
  updatedAt?: string
}

export interface GuideCourseListResponse {
  data: GuideCourse[]
}

export interface CreateGuideCourseRequest {
  slug: string
  title: string
  description?: string
  children: CourseNode[]
  sort_order?: number
  is_published?: boolean
}

export interface UpdateGuideCourseRequest {
  slug: string
  title: string
  description?: string
  children: CourseNode[]
  sort_order?: number
  is_published: boolean
}

export type LessonBlockType = LessonBlock['type']

export const BLOCK_TYPE_LABELS: Record<LessonBlockType, string> = {
  title: 'Sarlavha',
  description: 'Matn',
  video: 'Video',
  image: 'Rasm',
  link: 'Havola',
  file: 'Fayl',
}
